### DISCUSSION TITLE
Where to put util/style directives

### DISCUSSION THREAD

**Discussion by owensgit** [Legal-and-General/canopy#291] [Ideas] [answered]
We currently have the `lgMargin` and `lgPadding` directives to override margin/padding styles on an element, which currently sit in _**src/lib/spacing**_.

In the works we have an `lgVariant` directive to apply variant styling, such as `info`, `success` etc, and there are more planned, such as a fonts, breakpoints ([showAt](https://github.com/Legal-and-General/canopy/discussions/279)) etc.

Should we consolidate these into their own directory? What should this be called?

Also - should there be a module to contain all of these, or should we instead promote tree-shaking and only allow individual imports?

Keep in mind that we still have other non utility/style directives, such as the button component directive etc.

**Comment by owensgit** [comment, marked as answer]
My suggestion is a new directory called **util-directives**, for example

```
src/
  lib/
    util-directives/
      spacing/
      variant/
      font/
      showAt/
```

I would advocate **not** having a module to consolidate them, but instead require developers to import them directly to utilise tree-shaking.

**Reply by andy-polhill** [reply]
I quite the `util-directives`, it won't affect consumers of course but it might provide some structure as the codebase continues to grow. 

I agree on the no module part, having recently deprecated the forms one.

**Reply by elenagarrone** [reply]
I agree with all of that 👍  `util-directives` and no module.
